### DISCUSSION TITLE
Trivy not scanning Licenses for SBOM

### DISCUSSION THREAD

**Discussion by Osereme7** [aquasecurity/trivy#7001] [Q&A]
### Question

I am scanning some repositories written in golang which I cloned into my computer, I have trivy installed, when I scan the repo, Licenses shows results as :NONE, I want to know what packages are using what Licenses and Trivy does not give that, How can I make this work and what command should I use?

### Target

SBOM

### Scanner

License

### Output Format

SPDX

### Mode

Standalone

### Operating System

Ubuntu

### Version

```bash
Version: 0.52.1
Vulnerability DB:
  Version: 2
  UpdatedAt: 2024-06-11 12:12:15.151838022 +0000 UTC
  NextUpdate: 2024-06-11 18:12:15.151837731 +0000 UTC
  DownloadedAt: 2024-06-11 17:56:30.921841721 +0000 UTC
Java DB:
  Version: 1
  UpdatedAt: 2024-06-13 01:03:26.782257844 +0000 UTC
  NextUpdate: 2024-06-16 01:03:26.782257723 +0000 UTC
  DownloadedAt: 2024-06-13 16:02:31.288250534 +0000 UTC
```

**Comment by DmitriyLewen** [comment]
Hello @Osereme7 
Thanks for your interest to Trivy!

`go.mod` and `go.sum` files don't contain information about licenses.
We parse `$GOPATH/pkg/mod` directory to detect licenses from library directories (see https://aquasecurity.github.io/trivy/v0.52/docs/coverage/language/golang/#go-modules)

Regards, Dmitriy

**Reply by davidsuryaputra** [reply]
Hi @DmitriyLewen ,

Based on the documentation it is clear that for go lang we need to download the library first in order to scan the license.

But, i'm not clear with the NodeJS part especially Yarn section https://aquasecurity.github.io/trivy/v0.55/docs/coverage/language/nodejs/

Currently, in my repo some using Yarn 1 some using Yarn 2. My question is how to perform License scan? 
do I need to download the library using `yarn install` first, only then it will show the license scan result.

For npm section, it is clear that we need to download the library first.

Thanks,
David

**Reply by DmitriyLewen** [reply]
hello @davidsuryaputra 

> Trivy analyzes .yarn (Yarn 2+) or node_modules (Yarn Classic) folder next to the yarn.lock file to detect licenses.

It means that `.yarn`|`node_modules` directory should contain all related packages (as you correct said you need to use `yarn install` command for that).

**Reply by davidsuryaputra** [reply]
hi @DmitriyLewen 

Just want to check is the above behavior (download the library first) before performing the scan also application for vuln scan? 

if yes, that's means if we want to perform vuln scan to find vulnerable package/library, we must install the library first. is my understanding correct?

thanks

**Reply by DmitriyLewen** [reply]
For vulnerability scanning, Trivy only checks `package.json` present next to `yarn.lock`
This `package.json` is needed to detect Direct/Indirect dependencies and mark dev dependencies.

`node_modules`/`.yarn` dir is not required to scan vulnerabilities.

**Reply by davidsuryaputra** [reply]
Hi @DmitriyLewen thanks for your explanation. 

I have question related to Trivy scan, could you please help to check herehttps://github.com/aquasecurity/trivy/discussions/7633

Thanks
